src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php

```diff
@@ -62,6 +62,11 @@ public function __construct(ChoiceListFactoryInterface $choiceListFactory = null
             )
         );
 
+        if (null !== $translator && !$translator instanceof TranslatorInterface) {
```

Comment:
What about adding an argument type hint instead?

Reply:
Good idea, I applied that change.

Reply:
Turned out not to be that good, Fabien requested to get back to old fashioned, backwards compatible type check